modules/light-clients/08-wasm/types/client_state_test.go

```diff
@@ -601,258 +602,100 @@ func (suite *TypesTestSuite) TestVerifyMembershipGrandpa() {
-// 				// change the inserted proof
-// 				proof = []byte{}
-// 			}, false,
-// 		},
-// 	}
-
-// 	for _, tc := range testCases {
-// 		suite.Run(tc.name, func() {
-// 			suite.SetupWasmTendermint() // reset
-// 			testingpath = ibctesting.NewPath(suite.chainA, suite.chainB)
-// 			testingpath.SetChannelOrdered()
-// 			suite.coordinator.Setup(testingpath)
+					bz, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
 
-// 			// reset time and block delays to 0, malleate may change to a specific non-zero value.
-// 			delayTimePeriod = 0
-// 			delayBlockPeriod = 0
+					return &wasmvmtypes.Response{Data: bz}, types.DefaultGasUsed, nil
+				})
+			},
+			nil,
+		},
+		{
+			"wasm vm returns invalid proof error",
+			func() {
+				proof = []byte("invalid proof")
 
-// 			// create default proof, merklePath, and value which passes
-// 			// may be overwritten by malleate()
-// 			key := host.FullClientStateKey(testingpath.EndpointB.ClientID)
-// 			merklePath := commitmenttypes.NewMerklePath(string(key))
-// 			path, err = commitmenttypes.ApplyPrefix(suite.chainB.GetPrefix(), merklePath)
-// 			suite.Require().NoError(err)
+				suite.mockVM.RegisterSudoCallback(types.VerifyMembershipMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, sudoMsg []byte, store wasmvm.KVStore,
+					goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					return nil, types.DefaultGasUsed, commitmenttypes.ErrInvalidProof
+				})
+			},
+			commitmenttypes.ErrInvalidProof,
+		},
+		{
+			"proof height greater than client state latest height",
+			func() {
+				proofHeight = clienttypes.NewHeight(1, 100)
+			},
+			ibcerrors.ErrInvalidHeight,
+		},
+		{
+			"invalid path argument",
+			func() {
+				path = ibcmock.KeyPath{}
+			},
+			ibcerrors.ErrInvalidType,
+		},
+	}
 
-// 			proof, proofHeight = suite.chainB.QueryProof(key)
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupWasmWithMockVM()
 
-// 			clientState := testingpath.EndpointB.GetClientState().(*types.ClientState)
-// 			value, err = suite.chainB.Codec.MarshalInterface(clientState)
-// 			suite.Require().NoError(err)
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
 
-// 			tc.malleate() // make changes as necessary
+			path = commitmenttypes.NewMerklePath("/ibc/key/path")
+			proof = []byte("valid proof")
+			proofHeight = clienttypes.NewHeight(0, 1)
+			value = []byte("value")
 
-// 			clientState = testingpath.EndpointA.GetClientState().(*types.ClientState)
+			tc.malleate()
 
-// 			ctx := suite.chainA.GetContext()
-// 			store := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(ctx, testingpath.EndpointA.ClientID)
+			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)
```

Comment:
```suggestion
			clientStore := suite.store
```
You could remove `clientStore` because of [this](https://github.com/cosmos/ibc-go/blob/58584fcc87bdf35a2e2b502b6284098492fc545b/modules/light-clients/08-wasm/types/types_test.go#L84-L85)

Reply:
I'm not a fan either. But I think if they are there, we should use them